signer/threshold_validator.go

```diff
@@ -75,11 +77,27 @@ func NewThresholdValidator(
 }
 
 func (pv *ThresholdValidator) SaveLastSignedState(signState SignStateConsensus) error {
-	return pv.lastSignState.Save(signState, &pv.lastSignStateMutex, true)
+	return pv.lastSignState.Save(signState, &pv.lastSignStateMutex, &pv.asyncWaitGroup)
 }
 
 func (pv *ThresholdValidator) SaveLastSignedStateInitiated(signState SignStateConsensus) error {
-	return pv.lastSignStateInitiated.Save(signState, &pv.lastSignStateInitiatedMutex, true)
+	return pv.lastSignStateInitiated.Save(signState, &pv.lastSignStateInitiatedMutex, &pv.asyncWaitGroup)
+}
+
+// Stop safely shuts down the ThresholdValidator.
+func (pv *ThresholdValidator) Stop() {
+	pv.waitForSignStatesToFlushToDisk()
+}
+
+// waitForSignStatesToFlushToDisk waits for any sign states to finish writing to disk.
+func (pv *ThresholdValidator) waitForSignStatesToFlushToDisk() {
+	pv.asyncWaitGroup.Wait()
```

Comment:
`asyncWaitGroup` doesn't seem like a specific enough name given this is where the wait occurs. `pendingDiskWG` perhaps?

Reply:
renamed